components/display/include/display.hpp

```diff
@@ -256,36 +264,48 @@ class Display : public BaseComponent {
    * @brief Return the number of bytes that vram() can hold.
    * @return size_t Number of bytes that fit in the display buffer.
    */
-  size_t vram_size_bytes() const { return display_buffer_px_size_ * sizeof(lv_color_t); }
+  size_t vram_size_bytes() const { return display_buffer_px_size_ * sizeof(Pixel); }
 
 protected:
+  /**
+   * @brief LVGL event handler.
+   */
+  static void event_cb(lv_event_t *event) {
+    if(event->code == LV_EVENT_RESOLUTION_CHANGED) {
+      auto rotation = lv_display_get_rotation(lv_display_get_default());
+      auto rotation_callback = reinterpret_cast<rotation_fn>(lv_event_get_user_data(event));
+      rotation_callback(static_cast<DisplayRotation>(rotation));
```

Comment:
I know the `event_cb` is only registered in `init` if the callback function is not nullptr, but that may change in the future (e.g. if we support adding a rotation callback setter later), so best to go ahead and check that it's not nullptr here before calling it :)

Reply:
You're right, that's indeed better. Fixed!